p2p/stream/protocols/sync/client.go

```diff
@@ -0,0 +1,396 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	protobuf "github.com/golang/protobuf/proto"
+	"github.com/harmony-one/harmony/core/types"
+	syncpb "github.com/harmony-one/harmony/p2p/stream/protocols/sync/message"
+	sttypes "github.com/harmony-one/harmony/p2p/stream/types"
+	"github.com/pkg/errors"
+)
+
+// GetBlocksByNumber do getBlocksByNumberRequest through sync stream protocol.
+// Return the block as result, target stream id, and error
+func (p *Protocol) GetBlocksByNumber(ctx context.Context, bns []uint64, opts ...Option) ([]*types.Block, sttypes.StreamID, error) {
+	if len(bns) > GetBlocksByNumAmountCap {
+		return nil, "", fmt.Errorf("number of blocks exceed cap of %v", GetBlocksByNumAmountCap)
+	}
+
+	req := newGetBlocksByNumberRequest(bns)
+	resp, stid, err := p.rm.DoRequest(ctx, req, opts...)
+	if err != nil {
+		// At this point, error can be context canceled, context timed out, or waiting queue
+		// is already full.
+		return nil, stid, err
+	}
+
+	// Parse and return blocks
+	blocks, err := req.getBlocksFromResponse(resp)
+	if err != nil {
+		return nil, stid, err
+	}
+	return blocks, stid, nil
+}
+
+// GetEpochState get the epoch block from querying the remote node running sync stream protocol.
+// Currently, this method is only supported by beacon syncer.
+// Note: use this after epoch chain is implemented.
```

Comment:
This is only used for sync'ing epoch chain right? it shouldn't be needed more normal blocks sync

Reply:
Right. Let me remove this.

Reply:
Code removed